include/fmt/ranges.h

```diff
@@ -414,20 +417,86 @@ struct range_formatter<
     if (it != end && *it == 'n') {
       set_brackets({}, {});
       ++it;
+    } else {
+      bool check_for_s = false;
+      if (it != end && *it == '?') {
+        ++it;
+        detail::maybe_set_debug_format(underlying_, true);
+        set_brackets({}, {});
+        check_for_s = true;
+        is_debug = true;
+      }
+      if (it != end && *it == 's') {
+        if (!std::is_same<T, Char>::value) {
+          report_error("invalid format specifier");
+        }
+        if (!is_debug) {
+          set_brackets(detail::string_literal<Char, '"'>{},
+                       detail::string_literal<Char, '"'>{});
+        }
+        check_for_s = false;
+        is_string_format = true;
+        ++it;
+      }
+      if (check_for_s) {
+        report_error("invalid format specifier");
+      }
```

Comment:
There should be a check that the element is of `Char` type.

Reply:
I believe the check for element of `Char` type on line 430 is enough because in both the `s` and `?s` cases, the parser will pass through that check, unless I am missing something?